### ISSUE TITLE
[BUG] - Grafana Dashboard Link from the Rancher UI broken even after applying PR [harvester-installer/pull/666]

### ISSUE THREAD

**Issue by stalin4suse** [harvester/harvester#5389]
**Describe the bug**
Grafana dashboard broken even after applying [PR](https://github.com/w13915984028/harvester-installer/blob/fix5236/pkg/config/templates/patch/rancher-monitoring/102.0.0%2Bup40.1.2/nginx-config.yaml). 

**To Reproduce**
1. Update the `grafana-nginx-proxy-config` configmap in the `cattle-monitoring-system` on the harvester cluster with data from the `map $http_referer $final_appSubUrl` block mentioned in the following [PR](https://github.com/w13915984028/harvester-installer/blob/fix5236/pkg/config/templates/patch/rancher-monitoring/102.0.0%2Bup40.1.2/nginx-config.yaml). 
```
      map $http_referer $final_appSubUrl {
        ~.*/k8s/clusters/(c-m-.+)/api/v1/namespaces/cattle-monitoring-system/.*    '"appSubUrl":"/k8s/clusters/$1/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        ~.*/dashboard/harvester/c/(c-m-.+)/kubevirt.io.virtualmachine/.*/.*    '"appSubUrl":"/k8s/clusters/$1/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        ~.*/dashboard/harvester/c/(c-m-.+)/harvesterhci.io.addon/cattle-monitoring-system/.*    '"appSubUrl":"/k8s/clusters/$1/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        ~.*/dashboard/harvester/c/(c-m-.+)/.*      '"appSubUrl":"/k8s/clusters/$1/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        ~.*/dashboard/c/(c-m-.+)/.*     '"appSubUrl":"/k8s/clusters/$1/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        ~.*/api/v1/namespaces/cattle-monitoring-system/services/.*    '"appSubUrl":"/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
        default    '"appSubUrl":"/k8s/clusters/local/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy"';
      }
```
2. Login to Rancher --> Click the `Hamburger Menu` --> `Virtualization Management` --> `Manage` button of the Harvester cluster -->  `Explore` button --> `Workloads` --> `Pods` --> Delete the `rancher-monitoring-grafana` pod from the `cattle-monitoring-system` namespace to spin up a new pod. 

![image](https://github.com/harvester/harvester/assets/92844430/03fda101-e8b0-40ae-ba71-9ff3d373cefc)

3. Logout and Login to Rancher --> Click the `Hamburger Menu` --> `Virtualization Management` --> `Manage` button of the Harvester cluster --> `Explore` button --> Click on `Grafana` button.

![image](https://github.com/harvester/harvester/assets/92844430/d4f9bdb3-1109-4581-8159-d357a606682f)

4. Clicking on the link opens a new tab to Grafana with broken dashboard.

![image](https://github.com/harvester/harvester/assets/92844430/4b5707d6-a6d8-4748-b45e-6cc29991577c)

5. Click on `home dashboard` link and then `Sign In` button throws the following error message:

![image](https://github.com/harvester/harvester/assets/92844430/366f9b08-3a9e-4222-b8dc-be2bfa194424)

![image](https://github.com/harvester/harvester/assets/92844430/1dcc3945-a43e-48b4-a9fb-4a08ce407eca)

```json
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "services \"rancher-monitoring-grafana\" not found",
  "reason": "NotFound",
  "details": {
    "name": "rancher-monitoring-grafana",
    "kind": "services"
  },
  "code": 404
}
```

--

**Expected behavior**

- Clicking on the Grafana page should not display a broken dashboard but instead pull metrics from the harvester cluster and display them.

**Support bundle**
[harvester-supportbundle_1ea678f1-4959-40f8-b4b0-98313a5360f3_2024-03-18T11-14-17Z.zip](https://github.com/harvester/harvester/files/14635350/harvester-supportbundle_1ea678f1-4959-40f8-b4b0-98313a5360f3_2024-03-18T11-14-17Z.zip)

**Environment**
 - Harvester ISO version: v1.2.1
 - Rancher 2.7.6
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal with Equinix

**Additional context**
- Before clicking on the `Grafana` link, if you open browser `Inspect` option and search for `noreferrer` string in the `Element` tab. 
- You would see something similar to this: 
```
<a data-v-10dea0a6="" href="/k8s/clusters/c-m-nfhk4zss/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/d/rancher-cluster-nodes-1/rancher-cluster-nodes?orgId=1&amp;from=now-5m&amp;to=now&amp;refresh=30s&amp;theme=dark" target="_blank" rel="noopener noreferrer nofollow" previewlistener="true">Grafana <i data-v-10dea0a6="" class="icon icon-external-link"></i></a>
```
- When you hover your mouse on top of this link, it should highlight the `Grafana` button.
- Remove the `noreferrer` string from the URL in the `Elements` page and then click on the `Grafana` link and the dashboard would work fine. 

## Before
![image](https://github.com/harvester/harvester/assets/92844430/195c271d-8d80-4dd6-ad49-a6bafaac3f75)

## After
![image](https://github.com/harvester/harvester/assets/92844430/850cbab0-a961-448c-aa97-27651b414841)

**Comment by w13915984028** [comment]
@torchiaf

For the Rancher UI, the link to `Grafana` needs to remove the `noreferrer` as well, just like the Harvester dashboard `Grafana` link.  Could you help check is this an Rancher UI bug or Harvester UI bug, thanks.

`Rancher --> Click the Hamburger Menu --> Virtualization Management --> Manage button of the Harvester cluster --> Explore button --> Click on Grafana button`

**Comment by torchiaf** [comment]
@w13915984028 @stalin4suse What version are you using to test the issue ? 
The `noreferrer` was removed in UI version `v2.7.7`  and it works correctly: https://github.com/rancher/dashboard/blob/51664c1a194adde14e9b6493b1ae3255facbe815/shell/components/GrafanaDashboard.vue#L271.
Your screenshot shows `v2.7.6` in the bottom-left corner.
Could you double check please ?

**Comment by bk201** [comment]
@torchiaf I moved the milestone to v1.4.0 and created backport issues for tracking purposes, feel free to close backport issues if not needed.

**Comment by stalin4suse** [comment]
@torchiaf - The Rancher version is indeed `v2.7.6`. Made a typo and have corrected that. Will test the fix with Rancher `v2.7.7` and harvester `v1.2.1` and will share the updates.

**Comment by stalin4suse** [comment]
@torchiaf - I was able to test the fix with `v2.7.7` and it works. Thanks!

**Comment by torchiaf** [comment]
@stalin4suse @bk201 should this is issue and its back-ports be closed ?
https://github.com/harvester/harvester/issues/5422
https://github.com/harvester/harvester/issues/5423

**Comment by bk201** [comment]
Fixed in Rancher 2.7.7
